[IV-24-25] Objetivo 3

lmchaves opened the pull request:
* [x] ¿Incluye el fichero `iv.yaml` las claves de todos los objetivos
anteriores?
* [x] ¿Se ha desarrollado el fichero para las tareas a partir de un
   issue?
* [x] ¿Ese issue está asignado a un milestone?
* [x] ¿El milestone incluye, entre sus productos mínimamente viables, la
  existencia de herramientas que permitan automatizar tareas?
* [x] ¿Hemos justificado correctamente en caso de elegir un gestor de
   dependencias el que se use como gestor de tareas?
* [x] ¿El PR, que también es un issue, está asignado al mismo milestone?
* [x] ¿La tarea `check` efectivamente comprueba la sintaxis de los fuentes
  existentes y sólo eso?
* [x] ¿Se ha comprobado que las instrucciones incluidas en el `README.md` son
completas y correctas?

lmchaves committed:
añado las descripción de las clases

lmchaves committed:
feat #21: agrego el documento con los distintos gestores

lmchaves committed:
feat #21: agrego el documento con los distintos gestores

lmchaves committed:
feat #21: agrego lo necesario apra usar pdm

lmchaves committed:
feat #21: agrego lo necesario apra usar pdm

lmchaves committed:
feat #22: agrego el documento con los distintias opciones para automatizar las tareas

lmchaves committed:
feat #22: agrego el documento con los distintias opciones para automatizar las tareas

lmchaves committed:
feat #22:agrego el makefile necesario

lmchaves committed:
feat #22:agrego el makefile necesario

lmchaves committed:
Update iv.yaml

Agrego las claves necesarias

lmchaves committed:
Merge branch 'master' into Objetivo-3

lmchaves committed:
Update README.md

añado los enlaces a los documentos adicionales

lmchaves committed:
Update README.md

Agrego la instaruccion para isntalar

lmchaves committed:
Update README.md

lmchaves committed:
Update README.md

Añado la orden de comprobación

lmchaves committed:
Update README.md

Carlosmapego8 commented on docs/gestor_dependencias.md:
```diff
@@ -0,0 +1,39 @@
+
+
+
+Elegiría PDM por su rapidez y simplicidad, aunque varias cumplen el estándar PEP 621. Aunque Poetry destaca en documentación, es más lento, y Hatch tiene menos recursos. Pipenv está desactualizado, y UV, aunque prometedor, aún es inmaduro.
```
Te dejo un extracto de los errores frecuentes
>No establecer de antemano cuales son los criterios de elección de las diferentes herramientas.

Realizas la elección, eso es cierto, pero no con unos criterios previamente establecidos

lmchaves replied:
Aquí esta solucionado: [39ff4ea](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/39ff4ea21a375cc1c91f12f7b18aa47cf1e91e11)

Carlosmapego8 commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,31 @@
+
+Just es una herramienta de automatización de tareas que permite definir comandos de manera sencilla a través de un archivo justfile, similar a Make, Just es más fácil de escribir y entender (lenguaje más "cotidiano" ).  Es independiente del lenguaje de programación, lo que permite integrarlo en proyectos de Python y otros lenguajes sin problema. Make  tiene una comunidad más grande, lo que se tra ...
+
+Invoke es ideal para proyectos centrados exclusivamente en Python, pero Make te permite gestionar tareas en cualquier lenguaje, lo que lo hace más adecuado a largo plazo.
```
Lo mismo que en el anterior

lmchaves replied:
Aqui esta resuelto: [eb1e16f](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/eb1e16f6167fd85b2aed902976e2ff1b71277303)

Carlosmapego8 reviewed (changes requested):
Buenas, se me ha asignado revisión en tu PR. Te dejo un par de comentarios

lmchaves committed:
fix #21: agrego los criterios preestablecidos

lmchaves committed:
fix #22: agrego los criterios preestablecidos

lmchaves committed:
Resuelto conflicto en docs/gestor_dependencias.md y docs/gestor_tareas.md

lmchaves commented:
@Carlosmapego8 ya puedes revisar por si hay algo que todavía este mal

Carlosmapego8 reviewed (approved):
Por mi parte los veo bien @lmchaves, tienes mi aprobación

hossam1522 commented on README.md:
```diff
@@ -15,9 +15,17 @@ La clase **Sede** representa cada una de las sedes del taller, que pueden estar
 
+## Comprobación
+Para ver si la sintaxis es correcta ejcuta:
+make check check
```
typo

lmchaves replied:
Ya esta solucionado: [2b342ea](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/2b342eab19ad5e8ed60dc575f57c84e1aa5050b3)

hossam1522 commented on docs/gestor_dependencias.md:
```diff
@@ -0,0 +1,48 @@
+Para seleccionar el gestor de dependencias más adecuado para este proyecto, se han considerado los siguientes criterios:
+
+1. **Cumplimiento de estándares:** Preferencia por gestores que sigan el estándar PEP 621, garantizando mejores prácticas y compatibilidad futura.
+2. **Rendimiento:** Gestores rápidos y ligeros que no ralenticen el flujo de trabajo.
```
¿Realmente crees que es un criterio relevante? Me refiero, dudo que notes diferencias de rendimiento entre unos y otros. Además, ¿cómo lo medirías?

lmchaves replied:
Ya ha sido modificado [19e98aa](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/19e98aacb299ca97fe1a8c8048a8919e60595b47)

hossam1522 commented on docs/gestor_dependencias.md:
```diff
@@ -0,0 +1,48 @@
+
+1. **Cumplimiento de estándares:** Preferencia por gestores que sigan el estándar PEP 621, garantizando mejores prácticas y compatibilidad futura.
+2. **Rendimiento:** Gestores rápidos y ligeros que no ralenticen el flujo de trabajo.
+3. **Documentación y comunidad:** Herramientas con una comunidad activa y buena documentación para facilitar la resolución de problemas.
```
¿Cómo se valora si una documentación es buena o no?

lmchaves replied:
Me refiero a valorar considerando su claridad, cantidad de ejemplos prácticos, y facilidad para encontrar soluciones a problemas comunes(también influye la cantidad de recursos adicionales disponibles, como tutoriales, foros y comunidades activas)

hossam1522 commented on docs/gestor_dependencias.md:
```diff
@@ -0,0 +1,48 @@
+
+
+
+Elegiría PDM por su rapidez y simplicidad, aunque varias cumplen el estándar PEP 621. Aunque Poetry destaca en documentación, es más lento, y Hatch tiene menos recursos. Pipenv está desactualizado, y UV, aunque prometedor, aún es inmaduro.
```
Descartas, por ejemplo, Pipenv por estar desactualizado, pero no puedes descartar uno por algo que no esté entre los criterios.

lmchaves replied:
Ya lo he añadido: [5161cb5](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/5161cb554b5943fba117b039806183da36ab33cd)

hossam1522 commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,43 @@
+Para elegir el gestor de tareas más adecuado, se han considerado los siguientes criterios:
+
+1. **Facilidad de uso y configuración:** Herramientas sencillas de instalar y configurar, con una curva de aprendizaje accesible.
+2. **Compatibilidad con el lenguaje del proyecto:** Preferencia por herramientas integradas nativamente con Python o independientes del lenguaje.
```
¿No es demasiado obvio quiźas?

lmchaves replied:
Ya ha sido modificado [19e98aa](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/19e98aacb299ca97fe1a8c8048a8919e60595b47)

hossam1522 commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,43 @@
+
+1. **Facilidad de uso y configuración:** Herramientas sencillas de instalar y configurar, con una curva de aprendizaje accesible.
+2. **Compatibilidad con el lenguaje del proyecto:** Preferencia por herramientas integradas nativamente con Python o independientes del lenguaje.
+3. **Capacidades específicas:** Funcionalidades como ejecución remota, definición de tareas complejas o soporte para múltiples entornos.
```
¿Realmente necesitas alguna de las 3 cosas?

lmchaves replied:
Aqui esta solucionado: [19e98aa](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/19e98aacb299ca97fe1a8c8048a8919e60595b47)

hossam1522 commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,43 @@
+2. **Compatibilidad con el lenguaje del proyecto:** Preferencia por herramientas integradas nativamente con Python o independientes del lenguaje.
+3. **Capacidades específicas:** Funcionalidades como ejecución remota, definición de tareas complejas o soporte para múltiples entornos.
+4. **Comunidad y documentación:** Recursos disponibles para resolver problemas y aprender a utilizar la herramienta.
+5. **Escalabilidad y aplicabilidad a largo plazo:** Adecuación para proyectos de diferentes tamaños y lenguajes, con posibilidad de expansión futura.
```
No deberías pensar en nada relacionado con el largo plazo. Eso es deuda técnica, la cual se debe evitar.

lmchaves replied:
Aqui esta solucionado: [19e98aa](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/19e98aacb299ca97fe1a8c8048a8919e60595b47)

hossam1522 commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,43 @@
+
+## Paramiko
+
+Paramiko es una librería Python que proporciona una implementación del protocolo SSH. Permite la ejecución remota de comandos en sistemas a través de SSH y la transferencia se gura de archivos. Su uso esta orientado en aplicaciones que requieren interacciones seguras entre máquinas. Pero, al estar diseñado principalmente para la ejecución de tareas en entornos remotos, agrega una complejidad adic ...
```
typo: se gura

lmchaves replied:
ya esta arreglado [b72e7cc](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/b72e7ccb56c6a265e900fe6a3d65a34f54278758)

hossam1522 commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,43 @@
+
+Just es una herramienta de automatización de tareas que permite definir comandos de manera sencilla a través de un archivo justfile, similar a Make, Just es más fácil de escribir y entender (lenguaje más "cotidiano" ).  Es independiente del lenguaje de programación, lo que permite integrarlo en proyectos de Python y otros lenguajes sin problema. Make  tiene una comunidad más grande, lo que se tra ...
+
+Invoke es ideal para proyectos centrados exclusivamente en Python, pero Make te permite gestionar tareas en cualquier lenguaje, lo que lo hace más adecuado a largo plazo.
```
Lo comentado antes. Deuda técnica

lmchaves replied:
Aqui esta solucionado: [19e98aa](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/19e98aacb299ca97fe1a8c8048a8919e60595b47)

hossam1522 reviewed (commented):
Buenas! Se me ha asignado revisar tu issue, te dejo algunos comentarios.

lmchaves committed:
Update README.md

Corrijo "make check check"

wickeet commented on docs/gestor_dependencias.md:
```diff
@@ -0,0 +1,48 @@
+1. **Cumplimiento de estándares:** Preferencia por gestores que sigan el estándar PEP 621, garantizando mejores prácticas y compatibilidad futura.
+2. **Rendimiento:** Gestores rápidos y ligeros que no ralenticen el flujo de trabajo.
+3. **Documentación y comunidad:** Herramientas con una comunidad activa y buena documentación para facilitar la resolución de problemas.
+4. **Simplicidad de uso:** Gestores fáciles de instalar, configurar y usar, incluso para desarrolladores menos experimentados.
```
¿La simplicidad es algo objetivo o algo más bien basado en experiencia o conocimiento de herramientas similares?

lmchaves replied:
Aqui ha sido eliminado al ser un criterio subjetivo: [f213777](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/f213777b0a71340ccd1fa1bf0f3fc342e31f3eaf)

wickeet commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,43 @@
+3. **Capacidades específicas:** Funcionalidades como ejecución remota, definición de tareas complejas o soporte para múltiples entornos.
+4. **Comunidad y documentación:** Recursos disponibles para resolver problemas y aprender a utilizar la herramienta.
+5. **Escalabilidad y aplicabilidad a largo plazo:** Adecuación para proyectos de diferentes tamaños y lenguajes, con posibilidad de expansión futura.
+6. **Complejidad añadida:** Se penalizan herramientas que introducen más complejidad de la necesaria para el proyecto.
```
¿Qué supone una complejidad añadida? No creo que sea mal criterio pero igual podrías ser un poco más específico.

lmchaves replied:
Aqyui está expliacado mejor: [a2f9e40](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/a2f9e40c82cb6a1a63d76a14f8c8ba68eda6c57a)

lmchaves committed:
fix #21 corregimos algun cirterio cambiandolo por otro mas adecuado

wickeet commented on Makefile:
```diff
@@ -0,0 +1,32 @@
+		fi
+
+# Inicializar el proyecto si no tiene pyproject.toml
+init_project:
```
¿Esta comprobación es realmente necesaria? A no ser que el usuario final borre el archivo, algo que sería ponerse trabas a si mismo, o que tu quitases el archivo del repositorio (no tiene ningún sentido), el archivo va a estar ahí

lmchaves replied:
Si, no es necesaria [ba4d472](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/ba4d472c0cf1c6b115d53f818b7afb01f4f20f70)

wickeet reviewed (commented):
Los cambios que tienes que hacer son sencillos. ¡Ánimo que te falta poco!

lmchaves committed:
fix #22 modfico los criterios y cambio la expliación de la decisión final

lmchaves committed:
fix #22 arreglo typo: se gura

lmchaves committed:
fix #22: explico mejor tanto el critrio 1 como el 3

lmchaves committed:
fix #22 quito la dependencia init, innecesaria

lmchaves committed:
fix #21: elimno simpliciadad al ser un criterio subjetivo

lmchaves commented:
Gracias @hossam1522 y @wickeet ya he resuleto los comentarios

lmchaves commented:
@JJ ya esta listo para revisión

JJ commented on docs/gestor_dependencias.md:
```diff
@@ -0,0 +1,50 @@
+## Criterios de evaluación
+Para seleccionar el gestor de dependencias más adecuado para este proyecto, se han considerado los siguientes criterios:
+
+1. **Cumplimiento de estándares:** Preferencia por gestores que sigan el estándar PEP 621, garantizando mejores prácticas y compatibilidad futura.
```
PEP 621 es para los metadatos del proyecto; los que indican las dependencias son otros. En todo caso, enlázalos para que sea más fácil acceder a ellos.

lmchaves replied:
Aqui esta solucionado: [d373061](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/d37306170076c82fbdf7f86e08f034d6dd56a378)

JJ commented on docs/gestor_dependencias.md:
```diff
@@ -0,0 +1,50 @@
+Para seleccionar el gestor de dependencias más adecuado para este proyecto, se han considerado los siguientes criterios:
+
+1. **Cumplimiento de estándares:** Preferencia por gestores que sigan el estándar PEP 621, garantizando mejores prácticas y compatibilidad futura.
+2. **Documentación y comunidad:** Herramientas con una comunidad activa y buena documentación para facilitar la resolución de problemas. Será medido:
```
Explícitamente os digo que no es un buen criterio, tanto en el guión como en los errores frecuentes.

lmchaves replied:
Aqui esta solucionado: [d373061](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/d37306170076c82fbdf7f86e08f034d6dd56a378)

JJ commented on docs/gestor_dependencias.md:
```diff
@@ -0,0 +1,50 @@
+
+## Pipenv
+
+Combina **pip** y **env**, instala los paquetes mediante Pipfile.lock o Pipfile (no son estándares).
```
¿Qué estándar sigue?

lmchaves replied:
Aqui esta solucionado: [d373061](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/d37306170076c82fbdf7f86e08f034d6dd56a378)

JJ commented on docs/gestor_dependencias.md:
```diff
@@ -0,0 +1,50 @@
+
+Compatible con el estándar PEP 621, utilizando pyproject.toml para gestionar las dependencias, entornos virtuales y empaquetado de código.
+Es ligero y rápido, y enfocado en la simplicidad y el uso de archivos de bloqueo (pdm.lock) para garantizar versiones coherentes de las dependencias.
+Soporta múltiples entornos virtuales como Hatch, pero también permite una integración más sencilla con otros entornos como Docker.
```
¿Tiene que ver algo con los criterios? Si no lo hace, por favor no incluyáis información irrelevante para la elección o para la corrección.

lmchaves replied:
[d373061](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/d37306170076c82fbdf7f86e08f034d6dd56a378)

JJ commented on docs/gestor_dependencias.md:
```diff
@@ -0,0 +1,50 @@
+
+Es un gestor muy moderno (empezó en agosto de este año), sin embargo en muchos foros de Reddit recominedan migrar de gestores como poetry y hatch a uv.
+Un punto negativo es que no hay mucha documentación, debido al rápido ritmo de desarrollo de uv, por lo que si te surgen problemas, debes abrir un issue 
+(se puede ver como teinen mas de 1000 abiertos de hace minutos u horas, respondend bastante rápido).
```
typo: teinen

lmchaves replied:
Aqui esta arreglado: [d373061](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/d37306170076c82fbdf7f86e08f034d6dd56a378)

JJ commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,39 @@
+
+## Paramiko
+
+Paramiko es una librería Python que proporciona una implementación del protocolo SSH. Permite la ejecución remota de comandos en sistemas a través de SSH y la transferencia segura de archivos. Su uso esta orientado en aplicaciones que requieren interacciones seguras entre máquinas. Pero, al estar diseñado principalmente para la ejecución de tareas en entornos remotos, agrega una complejidad adici ...
```
¿Y esto qué tiene que ver con gestionar tareas?

lmchaves replied:
Aqui esta solucionado: [0e4affe](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/0e4affe8f1ea53e50ac80d823bc7d2de95b014a6)

JJ commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,39 @@
+
+## Fabric
+
+Fabric (basada en Paramiko) es una herramienta basada en Python para ejecutar comandos de shell de manera remota a través de SSH. Permite automatizar tareas en servidores y despliegues, extendiendo las funcionalidades de Invoke y Paramiko. Es una muy buena opción para gestionar tareas distribuidas en entornos remotos.Sin embargo, al igual que Paramiko, tiene una complejidad adicional innecesaria.
```
¿Es eso lo que buscamos? ¿De manera *remota*?

lmchaves replied:
Aqui esta solucionado: [0e4affe](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/0e4affe8f1ea53e50ac80d823bc7d2de95b014a6)

JJ commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,39 @@
+
+## Rake
+
+Rake está orientado principalmente a proyectos escritos en Ruby, lo que limita su utilidad para proyectos en Python y otros lenguajes. Aunque tiene una buena integración con Ruby, su comunidad y documentación son más pequeñas en comparación con Make, que tiene un soporte más amplio y recursos más accesibles.   
```
Por favor, mira los errores frecuentes.

lmchaves replied:
Aqui esta solucionado: [0e4affe](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/0e4affe8f1ea53e50ac80d823bc7d2de95b014a6)

JJ commented on Makefile:
```diff
@@ -0,0 +1,26 @@
+
+# Ejecutar flake8 para revisar el código
+check:
+		$(PDM_PATH) run flake8 src/
```
¿Esto *solo* comprueba la sintaxis?

lmchaves replied:
Aquí esta corregido: [98da6e5](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/98da6e50a37b9d594e004cb319af351bc98de9bb)

JJ reviewed (commented):
Por favor, mira también los errores frecuentes.

lmchaves committed:
fix #22: cambio flake por pylint que si que comprueba solo la sintaxis

lmchaves committed:
fix #21: se ha corregido en general el documetno

 Se ha mdoficado el PEP 621 por el correcto el 518

Corrijo los criterios, y eleimino aquella que directamente no sigue con el estándar

lmchaves committed:
fix #22: arreglo el docuemtno acorda al archivo de errores comunes

lmchaves commented:
@JJ ya está listo para revisión

JJ commented on Makefile:
```diff
@@ -17,9 +17,10 @@ install: install_pdm
 		$(PDM_PATH) run pytest
 
-# Ejecutar flake8 para revisar el código
+# Ejecutar pylint para revisar la sintaxis y estilo del código
```
Por favor, usad herramientas u opciones de comando que *solo* comprueben la sintaxis. Esto es un linter, diferente a flake8, que te aplica PEP8, pero también un linter (como su nombre indica)

lmchaves replied:
Ya lo he cambiado aqui: [48a789e](https://github.com/lmchaves/OrganizarTaller/pull/23/commits/48a789e4c87a0a898f47d55c406cd32845de06de)

JJ reviewed (changes requested):
Simplemente tratad de buscar  una herramienta u opción que compruebe la sintaxis.

lmchaves committed:
fix #22: elimno pylint y utilizo un comando que únicamente compruebe la sintaxis

lmchaves commented:
Revisión @JJ

JJ reviewed (approved):
👍
